.github/actions/silences-report-expired/silences-report-expired.sh

```diff
@@ -87,12 +96,26 @@ report() {
   local commit_sha="$3"
   local start_branch="$4"
   local repository_name="$5"
+  local mode="${6:-expired}"  # default is expired
 
   branch_name="clean-$file"
 
   # Map the git commit author to its github handle using github api
   userGithubHandle="$(gh api "/repos/${repository_name}/commits/${commit_sha}" -q '.author.login')"
 
+  # Check if silence has a linked open issue and skip if yes
+  issue_url="$($YQ e '.metadata.annotations.issue' "$directory/$file")"
+
+  if [[ -n "$issue_url" && "$issue_url" != "null" ]]; then
+    issue_number=$(echo "$issue_url" | grep -oE '[0-9]+$')
+    issue_state=$(gh issue view "$issue_number" --json state -q '.state' 2>/dev/null || echo "UNKNOWN")
+
+    if [[ "$issue_state" == "OPEN" ]]; then
+      echo "> skipping $file because linked issue #$issue_number is still open"
+      return 0
+    fi
+  fi
```

Comment:
We should not skip files because they have a linked issue. We still need to create pull request with the silence about to expire so people are notified and can act.
It is up to the human to decide if the silence should be kept and to check if the issue is still relevant or not, we should not decide for them here.